### PR TITLE
1955 - Adds additional field types to Compact Form

### LINKED ISSUE #1955: Compact Form: Implement additional features
**Is your feature request related to a problem? Please describe.**
The compact form implementation is currently just the foundation. Let's add some more feature support in.

**Describe the solution you'd like**
Support for:
- [ ] checkbox
- [ ] radios
- [x] date picker
- [x] time picker
- [x] file browser
- [x] lookup
- [ ] drop down

**Additional context**
We'll need to work through designs first

### PR DESCRIPTION
**Pull request by EdwardCoyle**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR adds styles and additional functionality for the following field types to the Compact Form:
- [x] Datepicker
- [x] Timepicker
- [x] Lookup
- [x] File Uploader

This PR also adds some better scaffolding to the DOM utility for dealing with parent elements, and some functional tests.  It also fixes a bug with columns in the compact form's RTL layout.

**Related github/jira issue (required)**:
Closes #1955 

**Steps necessary to review your pull request (required)**:
- Pull this branch and run the demoapp
- Navigate to http://localhost:4000/components/form-compact/example-index.html.  Check the new fields at the bottom and ensure they match the existing fields aesthetically.
- http://localhost:4000/components/form-compact/example-index.html?theme=uplift to see the Uplifted version
- Also check RTL: http://localhost:4000/components/form-compact/example-index.html?locale=ar-SA

**Included in this PR:**
- ✅An e2e or functional test for the bug or feature.
- ✅A note to the change log.

### CODE REVIEW

**Comment by davidcarlsonberg** [comment]
@EdwardCoyle I updated the last link in the Description to append a RTL locale. Originally it was `?theme=uplift`.

**Comment by EdwardCoyle** [comment]
@tmcconechy
- The blue dot does mean "required".  The compact form has a different design for required fields.
- The Lookup wasn't operational before, but I pushed some changes that make it so. 

@davidcarlsonberg, I pushed changes for both the Lookup field alignment and the left/right border placement on RTL.  Both of those should be corrected now.

File: src/components/form-compact/_form-compact.scss [hunk 1]
```diff
@@ -274,6 +274,11 @@ html[dir='rtl'] {
         border-right: 0;
       }
     }
+
+    // Font family changes on RTL affect the height of the FileUploader.
+    .fileuploader {
+      max-height: 29px;
```
Comment (davidcarlsonberg) [inline src/components/form-compact/_form-compact.scss]: This is fixed now on RTL but not on non-RTL where I believe 30px works.

Comment (EdwardCoyle) [reply src/components/form-compact/_form-compact.scss]: Seems like it works fine on Chrome, not on Firefox.

Comment (davidcarlsonberg) [reply src/components/form-compact/_form-compact.scss]: To my eye, and this seems crazy, Chrome works at 30px but Firefox works at 27.5px.

**Pull request merged**